### DISCUSSION TITLE
Issue with Twig Template Compilation in Winter CMS Application

### DISCUSSION THREAD

**Discussion by 01Kuzma** [wintercms/winter#942] [Q&A] [answered]
Hello,

I am currently collaborating with a colleague on the development of a project, and we have run into a problem with the compilation of Twig templates.

The specific issue we are facing is related to an "Unexpected character" error during the compilation process. The error message states: "Unexpected character '$' in "__string_template__606cb734c9c6c020d558f57e2523cf80" at line 1."

    Twig Template (plugins\kuzma\employees\views\mail\reminder_email.htm) is straight simple:

```
<p>{{ $message }}</p>
```

    Function in which the email is sent:

```
	private function sendReminderEmail(Employee $employee)
	{		
		$email = $employee->email;
		$subject = 'Subject In Cyrillic';
		$message = 'Message in In Cyrillic ' . $employee->get_date->format('Y-m-d') . '.';
		$message = htmlspecialchars($message);		

		try {
			Mail::to($email)->send(new ReminderEmail($subject, $message));
			$logMessage = sprintf(
				'Email sent for condition: %s to: %s <%s> at %s',
				$this->getReminderCondition($employee),
				$employee->name,
				$employee->email,
				now()->format('Y-m-d H:i:s')
			);
			Log::info($logMessage);
		} catch (\Exception $e) {
			$adminEmail = 'admin@admin.com';
			$adminSubject = 'error in Cyrillic';
			$adminMessage = 'message in Cyrillic ' . $employee->name;
			$adminMessage = htmlspecialchars($adminMessage);
			Mail::to($adminEmail)->send(new ReminderEmail($adminSubject, $adminMessage));
			Log::error(sprintf('Failed to send email to: %s <%s>. Error: %s', $employee->name, $employee->email, $e->getMessage()));
		}
	
	}

```

    Mailable Class (plugins\kuzma\employees\mail\ReminderEmail.php):

```
namespace Kuzma\Employees\Mail;

use Illuminate\Bus\Queueable;
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Mail\Mailable;
use Illuminate\Queue\SerializesModels;

class ReminderEmail extends Mailable implements ShouldQueue
{
    use Queueable, SerializesModels;

    public $subject;
    protected $message;

    public function __construct($subject, $message)
    {
        $this->subject = $subject;
        $this->message = $message;
    }

    public function build()
    {
        return $this->subject($this->subject)
            ->view('kuzma.employees::mail.reminder_email')
            ->with('message', $this->message);
    }	
	
}

```

We have reviewed the Twig template for any syntax errors or incorrect usage, but couldn't identify any obvious mistakes. It's just a one line.
The SMTP mailer is configured properly in the backed and it sents test mail.

Thank you for your attention to this matter. We greatly appreciate your expertise and prompt assistance in resolving this issue.

**Comment by mjauvin** [comment, marked as answer]
There is no "$" for variables in twig, just use `{{ message }}`

**Comment by 01Kuzma** [comment]
@mjauvin , oh my!
So many lines of code, and we were so blind with just this symbol 😄 
Thank you for your prompt reply!
